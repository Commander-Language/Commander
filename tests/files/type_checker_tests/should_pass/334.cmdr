//Test builtin sleep type checks
a = 1;
x = sleep(10);
x = sleep((1.0, true, 3)[a]);
