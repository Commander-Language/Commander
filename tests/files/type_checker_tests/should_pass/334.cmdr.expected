(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode a) SET (IntExprNode 1 INT) INT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode sleep (fn (INT) -> VOID)) (fn (INT) -> VOID)) (ExprsNode (IntExprNode 10 INT)) VOID) VOID))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode sleep (fn (INT) -> VOID)) (fn (INT) -> VOID)) (ExprsNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT))) VOID) VOID)))
